### PR TITLE
add a fusion op: fused_layernorm_residual_dropout_bias

### PR DESCRIPTION
**Pull request by zkh2016**
### PR types

 New features
### PR changes

OPs
### Describe

Fused elementwise_add, dropout, elementwise_add and layer_norm into one operator, only support Forward
```
//before fusion
out1 = elementwise_add(src, bias)
out2 = dropout(out1)
out3 = elementwise_add(residual, out2)
out = layer_norm(out3, other args)
```
```
//after fusion
out = fused_layernorm_residual_dropout_bias(src, residual, bias, other args)
```

### CODE REVIEW

File: paddle/fluid/operators/fused/fused_layernorm_residual_dropout_bias.h [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+limitations under the License. */
+
+#pragma once
+
+#include "paddle/fluid/operators/fused/fused_residual_dropout_bias.h"
+
+namespace paddle {
+namespace operators {
+
+namespace cg = cooperative_groups;
```
Comment (zhangting2020) [inline paddle/fluid/operators/fused/fused_layernorm_residual_dropout_bias.h]: 这个没用到？

File: paddle/fluid/operators/fused/fused_layernorm_residual_dropout_bias.h [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+  int idx = row_id * cols + col_id;
+  curandStatePhilox4_32_10_t state;
+  curand_init(seed, idx, increment, &state);
+
+  T factor = is_upscale_in_train ? static_cast<T>(1.0f / (1.0f - dropout_prob))
+                                 : factor = static_cast<T>(1.0f);
+  if (is_test) {
+    factor = is_upscale_in_train ? static_cast<T>(1.0f)
+                                 : static_cast<T>(1.0f - dropout_prob);
+  }
```
Comment (zhangting2020) [inline paddle/fluid/operators/fused/fused_layernorm_residual_dropout_bias.h]: 这段忘记替换为GetFactor了

**Comment by xingfeng01** [comment]
建议下个 PR 加些函数的注释，解释下计算逻辑。

File: paddle/fluid/operators/fused/fused_layernorm_residual_dropout_bias.h (hunk 2 above)
Comment (zkh2016) [reply paddle/fluid/operators/fused/fused_layernorm_residual_dropout_bias.h]: done

File: paddle/fluid/operators/fused/fused_layernorm_residual_dropout_bias.h (hunk 1 above)
Comment (zkh2016) [reply paddle/fluid/operators/fused/fused_layernorm_residual_dropout_bias.h]: done

**Review by xingfeng01 — APPROVED** [review]

**Comment by xingfeng01** [comment]
LGTM

**Review by zhangting2020 — APPROVED** [review]

**Review by lanxianghit — APPROVED** [review]

**Pull request merged**